azext_edge/tests/edge/conftest.py

```diff
@@ -83,3 +89,136 @@ def mocked_build_query(mocker, request):
     if request_params.get("return_value") is not None:
         build_query_mock.return_value = request_params["return_value"]
     yield build_query_mock
+
+
+# Int testing
+@pytest.fixture(scope="session")
+def settings():
+    from ..settings import DynamoSettings
+    try:
+        yield DynamoSettings()
+    except RuntimeError:
+        pytest.skip("Test requires resource group.")
+
+
+@pytest.fixture(scope="session")
+def tracked_resources():
+    resources = []
+    yield resources
+    for res in resources:
+        try:
+            run(f"az resource delete --id {res} -v")
+        except CLIInternalError:
+            logger.warning(f"failed to delete {res}")
+
+
+@pytest.fixture(scope="session")
+def tracked_keyvault(request, tracked_resources, settings):
+    from ..settings import EnvironmentVariables, convert_flag
+    settings.add_to_config(EnvironmentVariables.kv.value)
+    settings.add_to_config(EnvironmentVariables.skip_init.value, conversion=convert_flag)
+    if settings.env.azext_edge_skip_init:
+        # kv only needed for init for now
+        kv = None
+    elif settings.env.azext_edge_kv:
+        kv = run(f"az keyvault show -n {settings.env.azext_edge_kv} -g {settings.env.azext_edge_testrg}")
+    else:
+        run_id = id(request)
+        kv_name = f"opstestkv-{run_id}"
+        kv = run(f"az keyvault create -n {kv_name} -g {settings.env.azext_edge_testrg}")
+        tracked_resources.append(kv["id"])
+    yield kv
+
+
+@pytest.fixture(scope="session")
+def local_cluster_setup():
+    try:
+        run("kubectl api-resources")
+        yield
+        return
+    except CLIInternalError:
+        logger.info("Failed to access cluster. Trying to set up local cluster.")
+
+    try:
+        import os
+        os.environ["K3D_FIX_MOUNTS"] = "1"
+        run("kubectl cluster create -i ghcr.io/jlian/k3d-nfs:v1.25.3-k3s1")
+    except CLIInternalError:
+        logger.info("Failed to create cluster. Required for testing.")
+    yield
+    run("kubectl cluster delete")
+
+
+@pytest.fixture()
+def init_setup(request, tracked_resources, tracked_keyvault, local_cluster_setup, settings):
+    from ..settings import EnvironmentVariables
+    settings.add_to_config(EnvironmentVariables.cluster.value)
+    if settings.env.azext_edge_skip_init:
+        if not settings.env.azext_edge_cluster:
+            raise CLIInternalError("Cluster name required if not running init.")
+        yield {
+            "clusterName": settings.env.azext_edge_cluster,
+            "resourceGroup": settings.env.azext_edge_testrg
+        }
+        return
+
+    raise NotImplementedError(
+        "Please provide a valid cluster with AIO already deployed on it. The required parameters for "
```

Comment:
note to self: these are not actually required - need a good way to determine if cluster is there with AIO and then go from there